### ISSUE TITLE
Handle cuttlefish map errors

### ISSUE THREAD

**Issue by andrewjstone** [basho/clique#18]
Seeing the following using this branch: https://github.com/basho/riak_core/tree/feature/riak-cli-handoff-enable

```
Andrews-MacBook-Pro:riak_ee ajs$ dev/dev1/bin/riak-admin set handoff.disable_outbound=true
Invalid Configuration: {error,[{error,"Error transforming datatype for: handoff.disable_outbound"},
                               {error,"\"true\" is not a valid enum value, acceptable values are [\"on\",\"off\"]."}]}
```

We should display this information in a much nicer way.

**Comment by macintux** [comment]
Cuttlefish's odd approach to error handling makes it tempting to fix that first.

For example, why this:

``` erlang
cuttlefish_generator:minimal_map/2 -> {error,
                                                                 transform_datatypes,
                                                                 {error,
                                                                  [{error,
                                                                    "Error transforming datatype for: handoff.disable_outbound"},
                                                                   {error,
                                                                    "\"blah\" is not a valid enum value, acceptable values are [\"on\",\"off\"]."}]}}
```

Versus:

``` erlang
{error, transform_datatypes, {bad, "blah"}, {good, ["on", "off"]}}
```

**Comment by andrewjstone** [comment]
Fixed by #26
